exammanagement/main/models.py

```diff
@@ -1,3 +1,113 @@
+import uuid
+import jsonfield
 from django.db import models
+from django.urls import reverse
+from django.contrib.auth.models import User
+from django.utils.translation import gettext_lazy as _
 
 # Create your models here.
+
+
+class Genre(models.Model):
+    name = models.CharField(max_length=200, help_text=_(
+        'Genre of the subject (IT, Japanese, etc)'))
+
+    def __str__(self):
+        return self.name
+
+
+class Subject(models.Model):
+    name = models.CharField(
+        max_length=200, help_text=_('Title name of the subject'))
+    description = models.TextField(max_length=1000, help_text=_('Description'))
+    genres = models.ManyToManyField(
+        Genre, help_text=_('Select genre for this subject'))
+
+    def __str__(self):
+        return self.name
+
+    def get_absolute_url(self):
+        return reverse('subject-detail', args=[str(self.id)])
+
+    def display_genre(self):
+        """Create a string for the Genre. This is required to display genre in Admin."""
+        return ', '.join(genre.name for genre in self.genres.all()[:3])
+
+    display_genre.short_description = 'Genre'
+
+
+class Chapter(models.Model):
+    name = models.CharField(
+        max_length=200, help_text=_('Title name of the chapter'))
+    subject_id = models.ForeignKey(Subject, on_delete=models.CASCADE)
```

Comment:
set foreign_key thì ko cần hậu tố `_id` đâu e, khi make migration nó tự động thêm vào đới
Sửa tương tự bên dưới trong file nhé

```suggestion
    subject = models.ForeignKey(Subject, on_delete=models.CASCADE)
```